src/auth/digest/eDirectory/edir_ldapext.cc

```diff
@@ -368,33 +368,29 @@ static int nmasldap_get_simple_pwd(
 
     bufferLen = pwdBufLen = pwdLen+2;
     pwdBuf = (char*)SMB_MALLOC_ARRAY(char, pwdBufLen); /* digest and null */
-    if (pwdBuf == nullptr) {
+    if (!pwdBuf)
         return LDAP_NO_MEMORY;
-    }
 
     err = getLoginConfig(ld, objectDN, methodIDLen, &methodID, tag, &pwdBufLen, pwdBuf);
-    if (err == 0) {
-        if (pwdBufLen !=0) {
-            pwdBuf[pwdBufLen] = 0;       /* null terminate */
-
-            switch (pwdBuf[0]) {
-            case 1:  /* cleartext password  */
-                break;
-            case 2:  /* SHA1 HASH */
-            case 3:  /* MD5_ID */
-            case 4:  /* UNIXCrypt_ID */
-            case 8:  /* SSHA_ID */
-            default: /* Unknown digest */
-                err = LDAP_INAPPROPRIATE_AUTH;  /* only return clear text */
-                break;
-            }
-
-            if (!err) {
-                if (pwdLen >= pwdBufLen-1) {
-                    memcpy(pwd, &pwdBuf[1], pwdBufLen-1);  /* skip digest tag and include null */
-                } else {
-                    err = LDAP_NO_MEMORY;
-                }
+    if (err == 0 && pwdBufLen != 0) {
+        switch (pwdBuf[0]) {
+        case 1:  /* cleartext password  */
+            break;
+        case 2:  /* SHA1 HASH */
+        case 3:  /* MD5_ID */
+        case 4:  /* UNIXCrypt_ID */
+        case 8:  /* SSHA_ID */
+        default: /* Unknown digest */
+            err = LDAP_INAPPROPRIATE_AUTH;  /* only return clear text */
+            break;
+        }
+        if (!err) {
+            /* need (pwdBufLen-1) bytes for data + 1 for NUL => require >= pwdBufLen */
+            if (pwdLen >= pwdBufLen) {
+                memcpy(pwd, &pwdBuf[1], pwdBufLen - 1);
+                pwd[pwdBufLen - 1] = '\0';
```

Comment:
Official code explicitly claims to copy NUL byte to pwd already and, AFAICT, it does (assuming there is enough buffer space for that `pwdBuf[pwdBufLen] = 0` termination). This line is a better way to terminate `pwd` IMO, but I want to double check my understanding that this PR does not fix `pwd` termination. 

AFAICT, this PR fixes handling of passwords that are exactly pwdLen bytes long (or something along those lines) -- this `if` statement condition has changed -- pwdLen-byte passwords used to be rejected and now they are successfully returned. Is there a better way to describe this fix?